SHARE-478 GET /notifications/count

Kropiunig opened the pull request:
-implemented notification/count API
-removed old API
-replaced usages of old API with new API
-wrote test for API

---
### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroweb-Symfony/blob/develop/.github/contributing.md) and [wiki pages](https://github.com/Catrobat/catroweb-Symfony/wiki/) of this repository.

- [x] Include the name and id of the Jira ticket in the PR’s title eg.: `SHARE-666 The devils ticket`
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no warnings and errors 
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Verify that all tests are passing (CI), if not please state the test cases in the [section](#Tests) below
- [x] Perform a self-review of the changes
- [x] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] Put your ticket into the `Code Review` section in [Jira](https://jira.catrob.at/)
- [x] Post a message in the *#catroweb* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer
- [x] Check that your pull request has been successfully deployed to https://web-test-1.catrob.at/

### Additional Description
-implemented notification/count API
-removed old API
-replaced usages of old API with new API
-wrote test for API

Kropiunig committed:
SHARE-478 GET /notifications/count

-implemented notification/count API

Kropiunig committed:
SHARE-478 GET /notifications/count

-implemented notification/count API

Kropiunig committed:
SHARE-478 GET /notifications/count

-implemented notification/count API
-removed old API
-replaced usages of old API with new API
-wrote test for API

Kropiunig committed:
SHARE-478 GET /notifications/count

-implemented notification/count API
-removed old API
-replaced usages of old API with new API
-wrote test for API

Kropiunig committed:
SHARE-478 GET /notifications/count

-implemented notification/count API
-removed old API
-replaced usages of old API with new API
-wrote test for API

Kropiunig committed:
SHARE-478 GET /notifications/count

-implemented notification/count API
-removed old API
-replaced usages of old API with new API
-wrote test for API

dmetzner commented on assets/js/layout/sidebar.js:
```diff
@@ -1,59 +1,87 @@
-        if (badgeText === null) {
-          badge.innerHTML = (count <= maxAmountToFetch) ? count.toString() : (maxAmountToFetch + '+')
+  if (apiToCall === 'new') {
+    $.ajax({
```
Wieso hier jquery und unten das moderne fetch? :)
Würd dir nur empfehlen direkt überall die fetch api zu verwenden. https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch

Kropiunig committed:
SHARE-478 GET /notifications/count

-implemented notification/count API
-removed old API
-replaced usages of old API with new API
-wrote test for API

dmetzner commented on src/Api/Services/Notifications/NotificationsResponseManager.php:
```diff
@@ -3,7 +3,67 @@
+  {
+    $user->getFollowNotificationMentions();
+
+    $notifications_all = $this->notification_repository->findBy(['user' => $user]);
```
With a lot of notifications, this can easily become a huge performance overload. Every notification is fetched 1 by 1.
This could also be done by a single query.
